pkg/infrastructure/operator.go

```diff
@@ -54,3 +55,42 @@ func CheckOperatorExists(kubeCli *client.Client, namespace, operatorName string)
 
 	return true, nil
 }
+
+// CheckOperatorExistsUsingSubscription checks whether operator is existing and running. For this check informations from subscription are used.
+// If it is existing but not running, it returns true and an error
+func CheckOperatorExistsUsingSubscription(kubeCli *client.Client, namespace, subscriptionName string) (bool, error) {
+	log.Debugf("Checking if Operator with %s Subscription is deployed in namespace %s", subscriptionName, namespace)
+
+	subscription := &olmapiv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      subscriptionName,
+			Namespace: namespace,
+		},
+	}
+	if exists, err := kubernetes.ResourceC(kubeCli).Fetch(subscription); err != nil {
```

Comment:
I don't think that fetching only by name will do, since subscriptions can have generated names, that's why we fetch for `Package` and `CatalogSource` as well.

Won't be better to reuse this function? And externalize it to the `framework` (new with KOGITO-721 PR) package to be used by tests. They are very alike.

https://github.com/kiegroup/kogito-cloud-operator/blob/master/cmd/kogito/command/shared/operatorhub.go#L77

Reply:
Good point, will take a look on it.

Reply:
As this one was merged, created https://issues.redhat.com/browse/KOGITO-916 to follow-up on this (and assigned to you @sutaakar if you have time. If not I can have a look)